apps/backend/src/auth/auth.service.ts

```diff
@@ -0,0 +1,35 @@
+import { AuthSchProfile, BmeUnitScope } from '@kir-dev/passport-authsch';
+import { Injectable } from '@nestjs/common';
+import { JwtService } from '@nestjs/jwt';
+import { User } from '@prisma/client';
+import { PrismaService } from 'nestjs-prisma';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    private prisma: PrismaService,
+    private jwtService: JwtService
+  ) {}
+
+  async findOrCreateUser(userProfile: AuthSchProfile): Promise<User> {
+    const user = await this.prisma.user.findUnique({ where: { authSchId: userProfile.authSchId } });
+    if (user) return user;
+    return await this.prisma.user.create({
+      data: {
+        authSchId: userProfile.authSchId,
+        fullName: userProfile.displayName,
+        nickName: userProfile.firstName,
+        email: userProfile.email,
+        isSchResident: userProfile.bmeStatus.includes(BmeUnitScope.BME_VIK_ACTIVE),
```

Comment:
It think we need an isActiveVikStudent column in the enitity too

Reply:
do we even need these fields in the db? They will only ever be used when a registration period has ended, but then we will need to query authSch for up-to-date info, I don't see much use of storing this data.